libs/execution_context/ebpf_maps.c

```diff
@@ -264,6 +265,26 @@ _associate_program_with_prog_array_map(_In_ ebpf_core_map_t* map, _In_ const ebp
     return result;
 }
 
+static bool // Returns true if ok, false if not.
+_check_value_type(_In_ const ebpf_core_map_t* outer_map, _In_ const ebpf_object_t* value_object)
+{
+    if (outer_map->ebpf_map_definition.type != BPF_MAP_TYPE_ARRAY_OF_MAPS &&
+        outer_map->ebpf_map_definition.type != BPF_MAP_TYPE_HASH_OF_MAPS) {
+        return true;
+    }
+
+    ebpf_core_map_t* template = outer_map->inner_map_template;
+    ebpf_object_t* template_object = &template->object;
+    const ebpf_map_t* value_map = (ebpf_map_t*)value_object;
+
+    // TODO: what other map attributes need to match?
```

Comment:
File an issue for this so it doesn't get lost.

Reply:
Done